### PR TITLE
Podcast clipboard fix

### PR DESCRIPTION
**Pull request by MRJOHN5ON**
after careful examination of the resource from [Steven Boutcher](https://hashnode.com/@StevenBoutcher) [https://stevenqa.hashnode.dev/paste-text-from-your-clipboard-in-playwright-typescript](https://stevenqa.hashnode.dev/paste-text-from-your-clipboard-in-playwright-typescript). I was able to fix my inability to read the clipboard. 

With my limited understanding on Javascript, typescript and Playwright... this has encouraged me to learn more about `BrowserContext` method and `Navigator` object

Previously I was following this resource from [Butch Mayhew](https://playwrightsolutions.com/author/butch/) [https://playwrightsolutions.com/how-do-i-access-the-browser-clipboard-with-playwright/](https://playwrightsolutions.com/how-do-i-access-the-browser-clipboard-with-playwright/) which I still used in part of my helper function to help me read what was actually being stored in the clipboard `await page.evaluate(() => navigator.clipboard.readText());`

However I am still unsure how Butch was able to read and use the clipboard content..
After some research, I realized I needed to add a `return` statement. 
`const clipboardContent = await page.evaluate(() => navigator.clipboard.readText());
    return clipboardContent;`

### CODE REVIEW

File: tests/podcast.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { test, expect, } from '@playwright/test';
+import { copyWithButton, } from './helper';
+
+test('Choose any podcast. Verify the shareable link matches the text copied from the Copy Link button' , async ({ page, context }) => {
+
+    await page.goto('https://music.amazon.com/')
+    page.waitForLoadState('domcontentloaded')
```
Comment (steven-the-qa) [inline tests/podcast.spec.ts]: I tried locally, and it doesn't seem like you need a `waitForLoadState` check. By default, Playwright will actually wait longer than the `domcontentloaded` state (it waits til `load`
Screenshot 2024-12-16 at 9 38 52 PM
Screenshot 2024-12-16 at 9 41 23 PM
).

Here's some notes from Claude (Cursor) about that (and the [docs](https://playwright.dev/docs/api/class-frame#frame-wait-for-load-state) if you want):

File: tests/podcast.spec.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    // click on the first podcast shown
+    await page.locator('music-vertical-item').first().click();
+    await expect(page).toHaveURL(/https:\/\/music\.amazon\.com\/podcasts\/.+/)
+
+    //click the share button
+    const shareButton = page.getByTestId('detailHeaderButton2')
+    await shareButton.click();
+
+    const linkField = await page.locator('._3QpxCCZ2ZUyhnlmpwSU7as ').inputValue()
```
Comment (steven-the-qa) [inline tests/podcast.spec.ts]: This also works:

`const linkField = await page.getByTestId("dialog").locator("input").inputValue();`

## The parent div has an `id=dialog`
image

## Using the locator in a debug sesh
image

File: tests/podcast.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    expect(linkField).toContain('https://music.amazon.com/podcasts/')
+
+    //press the copy link button
+    const copyButton = page.locator('music-button').filter({ hasText: 'Copy link' }).getByRole('button')
```
Comment (steven-the-qa) [inline tests/podcast.spec.ts:23]: Nice 🚀 I tried to do better than this, but couldn't lol

File: tests/podcast.spec.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    // using the helper function read the clipboard content
+    const clipboardText = await copyWithButton(page, context, copyButton)
+    
+    // verify the clipboard content is the same as the link field
+    expect(clipboardText).toEqual(linkField)
+
+
+
```
Comment (steven-the-qa) [inline tests/podcast.spec.ts]: Empty lines 31-33 can be removed

File: tests/amazon.spec.ts [hunk 5]
```diff
@@ -252,4 +252,5 @@ test.describe('Adding to Cart', () => {
     });
 
 });
-
+/////
+///
```
Comment (steven-the-qa) [inline tests/amazon.spec.ts]: Remove these

File: playwright.config.ts [hunk 6]
```diff
@@ -10,12 +10,14 @@ export default defineConfig({
...
     trace: 'retain-on-failure',
     testIdAttribute: "id",
+    permissions: ["clipboard-read"]
   },
 
   projects: [
     {
       name: 'chrome',
       use: { ...devices['Desktop Chrome'] },
+
```
Comment (steven-the-qa) [inline playwright.config.ts:20]: Unnecessary extra line here.

Also, for your `package.json` file, add a debug script. It helped me when reviewing your PR, might help you save some keystrokes in the future. It goes in your `scripts` object:

`"debug": "npx playwright test --project=chrome --debug"`

**Review by steven-the-qa — APPROVED** [review]
Just added a few small suggestions. Overall you did great! Seems like you picked up some fun clipboard tricks along the way, too 📋

File: tests/podcast.spec.ts (hunk 2 above)
Comment (MRJOHN5ON) [reply tests/podcast.spec.ts]: I feel 100000000% better about this locator! Good find, my eyes overlooked this ID. 

Also I think I need a lesson on using the debugger.. maybe I'll find some youtube videos.... unless you teach it in the course? I remember the section on the trace viewer, but not the debugger.

**Pull request merged**